File: src/yl14/GameCoreAPI/utils/ChatChannel.php
<?php

namespace yl14\GameCoreAPI\utils;

use yl14\GameCoreAPI\GameCoreAPI;

class ChatChannel {

    private $plugin;

    private $channelName = null;
    private $players = []; /** @var \yl14\GameCoreAPI\utils\CustomPlayer */

    private $chatformat = "";
    private $isMute = false;

    public function __construct(GameCoreAPI $plugin, string $channelName, string $chatformat = "") {
        $this->plugin = $plugin;
        $this->channelName = $channelName;
        $this->chatformat = $chatformat;
    }

    public function addPlayer(CustomPlayer $player) {
        if(!isset($this->players[$player->getPlayer()->getName()])) {
            $player->setChatChannel($this);
            $this->players[$player->getPlayer()->getName()] = $player;
        }
    }

    public function removePlayer(CustomPlayer $player) {
        if(isset($this->players[$player->getPlayer()->getName()])) {
            $player->setChatChannel(null);
            unset($this->players[$player->getPlayer()->getName()]);
        }
    }

    public function removeAll() {
        foreach($this->players as $player) {
            $this->removePlayer($player);
        }
    }

    public function sendMessage(CustomPlayer $player, String $message) {
        if(!$this->isMute) {
            if($this->chatformat == "") {
                $this->broadcastMessage('[' . $player->getPlayer()->getName() . ']' . $message);
            } else {
                $this->broadcastMessage(str_replace(['PLAYERNAME', 'MESSAGE'], [$player->getPlayer()->getName()], $message));
            }
        }
    }

    public function broadcastMessage(String $message) {
        foreach($this->plugin->getServer()->getOnlinePlayers() as $player) {
            $player->sendMessage($message);
        }
    }

    public function setMute(bool $mute = false) {
        $this->mute = $mute;
    }
}

File: src/main/java/xyz/hjdmc/gamecoreapi/utils/ChatChannel.java
package xyz.hjdmc.gamecoreapi.utils;

import cn.nukkit.Player;
import xyz.hjdmc.gamecoreapi.GameCoreAPI;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

public class ChatChannel {

    private final GameCoreAPI plugin;
    private final String gameid;

    private HashMap<String, Player> players = new HashMap<String, Player>();
    private String format = "";
    private Boolean mute = false;

    public ChatChannel(GameCoreAPI plugin, String gameid) {
        this.plugin = plugin;
        this.gameid = gameid;
    }

    public void addPlayer(Player player) {
        if(!players.containsKey(player.getName())) {
            players.put(player.getName(), player);
            if(player.isOnline()) {
                this.plugin.getInGamePlayer(this.plugin, player).setChatChannel(this);
            }
        }
    }

    public void removePlayer(Player player) {
        if(players.containsKey(player.getName())) {
            players.remove(player.getName());
            if(player.isOnline()) {
                this.plugin.getInGamePlayer(this.plugin, player).removeChatChannel();
            }
        }
    }

    public void removeAllPlayers() {
        for(Map.Entry<String, Player> entry : players.entrySet()) {
            removePlayer(entry.getValue());
        }
    }

    public Player getPlayer(Player player) {
        return players.get(player.getName());
    }

    public Player[] getAllPlayer() {
        ArrayList<Player> playerList = new ArrayList<Player>();
        for(Map.Entry<String, Player> entry : players.entrySet()) {
            playerList.add(entry.getValue());
        }
        Player[] playerArray = new Player[playerList.size()];
        playerArray = playerList.toArray(playerArray);
        return playerArray;
    }

    public String getFormat() {
        return this.format;
    }

    public void setFormat(String format) {
        this.format = format;
    }

    public void setMute(Boolean mute) {
        this.mute = mute;
    }

    public String getGameid() {
        return this.gameid;
    }
}
